flink-kubernetes-operator-autoscaler/src/test/java/org/apache/flink/kubernetes/operator/autoscaler/AutoScalerFlinkMetricsTest.java

```diff
@@ -66,8 +67,10 @@ public void testMetricsRegistration() {
         initRecommendedParallelism(evaluatedMetrics);
         lastEvaluatedMetrics.put(resourceID, evaluatedMetrics);
 
-        metrics.registerScalingMetrics(() -> lastEvaluatedMetrics.get(resourceID));
-        metrics.registerScalingMetrics(() -> lastEvaluatedMetrics.get(resourceID));
+        metrics.registerScalingMetrics(
+                () -> List.of(jobVertexID), () -> lastEvaluatedMetrics.get(resourceID));
+        metrics.registerScalingMetrics(
+                () -> List.of(jobVertexID), () -> lastEvaluatedMetrics.get(resourceID));
```

Comment:
Could you help adding a test that some metrics are collected after `registerScalingMetrics`?

Reply:
Sure I can do that, although this is not the scope of this JIRA. The scope here is to register all metrics, not check that they are reported. There are already tests that do that.

Reply:
Sorry, Max. I didn't express clear.

This PR you fix a bug that some metrics aren't ready when  `registerScalingMetrics` is called, so these metrics cannot be registerd even if they are ready in the future.

Could we adding a test for this case? This test should fail without your PR, and it should success with your PR.

Please let me know if it's not clear, thanks~

Reply:
I added this test in https://github.com/apache/flink-kubernetes-operator/pull/660/commits/7cc8bbc51cbf3aeba9d035f50bf0d1b76e8b5e8f